src/Front/Parser.y

```diff
@@ -77,9 +79,13 @@ indeed semantically) distinct from actual let-bindings (@defLets@).
 They are separated with '||' rather than ';' because the latter somehow causes
 shift-reduce conflicts.
 -}
-defTop                                --> [Definition]
-  : defLet '||' defTop                  { $1 : $3 }
-  | defLet                              { [$1] }
+defTop                                --> [TopDef]
```

Comment:
This is kind of ugly. Why don't we separate this into:

```
defTops  --> [Definition]
  : defTop '||' defTops { $1 : $3 }
  | defTop { [$1] }
defTop --> Definition
  : defLet { TopDef $1 }
  | defClass { TopClass $1 }
  | defInst { TopInst $1 }
```

Something like that